### PR TITLE
Changelog Auto-Updater

### PR DESCRIPTION
**Pull request by Holzhaus**
This already allows automating the changelog updates via repository dispatch. I'll open a corresponding PR on the code repo. Please see the indivial commit messages for details.

To trigger a rebuild automatically, a PAT `REPO_TOKEN` needs to be present in the manual repo's secrets, because the default `GITHUB_TOKEN` doesn't to suffice.

### CODE REVIEW

**Comment by Holzhaus** [comment]
Here's an example:

1. Changelog Workflow: https://github.com/Holzhaus/manual/actions/runs/1101700527 (triggered from command line via cURL)
2. Commit created by the changelog workflow: https://github.com/Holzhaus/manual/commit/080084a00b65ef02b278157295fd52b48a98ebc0 (sorry for the large diff, because the Markdown to RST converter uses a different indentation style)
3. Rebuild (triggered by the changelog workflow): https://github.com/Holzhaus/manual/actions/runs/1101701820

**Comment by Holzhaus** [comment]
Who can add the repo secret? @daschuer maybe?

**Comment by daschuer** [comment]
> Who can add the repo secret? @daschuer maybe?

I had just a look in the settings, but I don't know what to do. Do you have some hints?

**Comment by Holzhaus** [comment]
Ready to merge. Tested on my own repo, here's the commit generated by the bot: https://github.com/Holzhaus/manual/commit/a4b9555f21ae0a6812883a9e3dea55653980b89e

**Comment by Holzhaus** [comment]
Merge?

graphics-spongebob-163388

**Comment by Holzhaus** [comment]
Ping. I suggest to merge this before mixxxdj/mixxx#4210 and mixxxdj/mixxx#4211 so that we can test if it works.

File: tools/update_changelog.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        f"{branch}/CHANGELOG.md"
+    )
+    r.raise_for_status()
+    return r.text
+
+
+def changelog_to_rst(changelog):
+    # Replace headline with "Version History"
+    changelog = re.sub(
+        "^# Changelog$",
```
Comment (Swiftb0y) [inline tools/update_changelog.py:39]:
```suggestion
        r"^#\s*Changelog\s*$",
```
makes it a bit more robust without producing false positives

**Comment by daschuer** [comment]
I think this will be hard to maintain later because of some hidden implications. 
I am not able to point them out, because I have not yet fully understand how it works. 
Would you mind to decorate the changes with some comments to improve the situation?

**Comment by daschuer** [comment]
This introduces some implications the author of the CHANGLOG.md file should be aware of. 
How can we make it sure that he his aware?

**Review by Swiftb0y — CHANGES_REQUESTED** [review]

**Comment by Holzhaus** [comment]
> I think this will be hard to maintain later because of some hidden implications.
> I am not able to point them out, because I have not yet fully understand how it works.
> Would you mind to decorate the changes with some comments to improve the situation?

I added some explaining comments, however I'm unsure if this is what you meant.

> This introduces some implications the author of the CHANGLOG.md file should be aware of.
> How can we make it sure that he his aware?

The implication is that `CHANGELOG.md` has to be valid markdown. This is not new. Anyway, to enforce valid markdown I opened https://github.com/mixxxdj/mixxx/pull/4222.

**Comment by Holzhaus** [comment]
Ping.

File: tools/update_changelog.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        r"[lp:\g<lpid>](https://bugs.launchpad.net/mixxx/+bug/\g<lpid>)",
+        changelog,
+    )
+
+    # Ensure that all GitHub PRs are hyperlinks
+    changelog = re.sub(
+        r"(?!\[)#\b(?P<ghid>\d+)\b(?!\])",
+        r"[#\g<ghid>](https://github.com/mixxxdj/mixxx/pull/\g<ghid>)",
+        changelog,
+    )
```
Comment (daschuer) [inline tools/update_changelog.py:57]: All these sub() calls are putting implications on the Changelog file.
Can we move the link check to the pre-commit hook of the main repro?
Can we remove the Changelog headline hack?
IMHO there is nothing bad about using the same headline everywhere.

**Review by daschuer — APPROVED** [review]
LGTM, thank you.

**Comment by daschuer** [comment]
@Swiftb0y: merge?

**Review by Swiftb0y — APPROVED** [review]
Sorry, I forgot I had to approve the PR after my suggestions have been incorporated.

**Comment by Swiftb0y** [comment]
I suggest we make an exception this time and let @Holzhaus merge both PRs himself in the order they require to be merged.

**Comment by Holzhaus** [comment]
It works: https://github.com/mixxxdj/manual/commit/52c82e0c567927ab98020f8cd576d6c487e37128

**Pull request merged**